packages/frontend/src/context/StoreContext.tsx

```diff
@@ -2,678 +2,156 @@
-  };
+  }, [relayClient]);
 
-  const commitOrder = async () => {
-    try {
-      if (!relayClient) throw Error(`Disconnected from relayClient`);
-      if (!selectedCurrency) throw Error(`No currency selected.`);
-      const checkout = await relayClient.commitOrder({
-        orderId: hexToBytes(orderId as OrderId),
-        currency: {
-          tokenAddr: hexToBytes(selectedCurrency!),
-          chainId: usedChainId,
-        },
-        payeeName: "default",
-      });
-      return {
-        requestId: bytesToHex(checkout.requestId),
-        orderFinalizedId: bytesToHex(checkout.orderFinalizedId),
-        error: null,
-      };
-    } catch (error) {
-      const errMsg = error as { message: string };
-      invalidateOrder(errMsg.message);
-      return { error: errMsg.message };
-    }
+  // useEffect(() => {
+  //   (async () => {
+  //     const currencies = Array.from([...acceptedCurrencies.keys()]);
+  //     const _cur = currencies.filter((a) => !acceptedCurrencies.get(a));
+  //     _cur.map(async (address) => {
+  //       const { symbol } = await getTokenInformation(address);
+
+  //       setAcceptedCurrencies({
+  //         type: UPDATE_SYMBOL,
+  //         payload: { tokenAddr: address, symbol },
+  //       });
+  //     });
+  //   })();
+  // }, [acceptedCurrencies]);
+
+  // const verify = async (
+  //   _orderItems: Map<OrderId, OrderState>,
+  //   _pubKeys: `0x${string}`[],
+  // ) => {
+  //   console.log(_orderItems, _pubKeys);
+  // const addresses = _pubKeys.map((k) => {
+  //   return Address.fromPublicKey(hexToBytes(k)).toString();
+  // });
+  // const keysArr: `0x${string}`[] = _orderItems.size
+  //   ? Array.from([..._orderItems.keys()])
+  //   : [];
+  // for (const _orderId of keysArr) {
+  //   const _order = _orderItems.get(_orderId) as OrderState;
+  //   if (_order && _order.status !== Status.Failed) {
+  //     const sig = _order.signature as `0x${string}`;
+  //     const retrievedAdd = await relayClient!.recoverSignedAddress(
+  //       _orderId,
+  //       sig,
+  //     );
+  //     if (addresses.includes(retrievedAdd.toLowerCase())) {
+  //       console.log("inside inclue", _orderId);
+  //       setOrderId(_orderId);
+  //     }
+  //   }
+  // }
+  // };
+
+  const getOrderId = async () => {
+    //FIXME: This part is still wip. We will have to go through the openOrders(array) and verify the signed address via keycard store. There should only be one open order per clerk.
+    // For now just set current order id as the first open order
+    const openOrders = await stateManager?.orders.getStatus(Status.Pending);
+    let order_id: OrderId;
+    if (openOrders && openOrders.length) {
+      order_id = openOrders[0] as OrderId;
+      setOrderId(order_id);
+    } else {
+      // If open order id does not exist, create new id
+      const { id } = await stateManager!.orders.create();
+      setOrderId(id);
+      order_id = id;
+    }
+    return order_id;
```

Comment:
I think orderId should live globally (in context) since we are going to have to map the open orders, get the signed address for the order, then key in that address in the KeyCard store.